### PR TITLE
C++14 compliant implementation of aligned_alloc

### PR DESCRIPTION
**Pull request by fabinsch**
- also fix the [[MAYBE_UNSUED]] problem which is introduced in c++17

with this PR, including also the changes done in #78, I can compile the proxsuite library enforcing c++14.

### CODE REVIEW

File: include/proxsuite/linalg/veg/memory/alloc.hpp [hunk 1]
```diff
@@ -52,6 +52,72 @@ aligned_alloc(std::size_t alignment, std::size_t size)
 } // namespace alignment
 #endif
 
+namespace alignment {
```
Comment (jcarpent) [inline include/proxsuite/linalg/veg/memory/alloc.hpp:55]: Could you provide the sources of this code?

File: include/proxsuite/helpers/common.hpp [hunk 2]
```diff
@@ -27,4 +27,12 @@ struct infinite_bound
 } // helpers
 } // proxsuite
 
+#if __cplusplus >= 201703L
```
Comment (jcarpent) [inline include/proxsuite/helpers/common.hpp:30]: Should be nice to move to a common file proxsuite/fwd.hpp if possible.

File: include/proxsuite/linalg/veg/memory/alloc.hpp [hunk 3]
```diff
@@ -52,6 +52,81 @@ aligned_alloc(std::size_t alignment, std::size_t size)
 } // namespace alignment
 #endif
 
+// Support aligned_alloc for c++14, code from boost/align.
+namespace alignment {
```
Comment (jcarpent) [inline include/proxsuite/linalg/veg/memory/alloc.hpp:56]: Could you also add an ifdef for C++ lower than C++17?

File: include/proxsuite/helpers/common.hpp (hunk 2 above)
Comment (fabinsch) [reply include/proxsuite/helpers/common.hpp]: Good idea, I created the file and moved the macro definitions there.

File: include/proxsuite/linalg/veg/memory/alloc.hpp (hunk 3 above)
Comment (fabinsch) [reply include/proxsuite/linalg/veg/memory/alloc.hpp:56]: I did so in the proxsuite/fwd.hpp

**Review by jcarpent — DISMISSED** [review]

**Review by jcarpent — APPROVED** [review]

**Pull request merged**